src/node_api.cc

```diff
@@ -508,9 +514,22 @@ void napi_module_register_cb(v8::Local<v8::Object> exports,
                              v8::Local<v8::Value> module,
                              v8::Local<v8::Context> context,
                              void* priv) {
+  v8::Isolate* isolate = context->GetIsolate();
   napi_module* mod = static_cast<napi_module*>(priv);
+
+  // Can we be absolutely sure that, if isolate->GetData() returns non-NULL, it
+  // means that the pointer returned is one of ours (i.e. a napi_env)? What if
+  // some other framework makes use of slot 0? There really seems to be a need
+  // for a more robust way of assigning arbitrary data to the isolate.
+  if (isolate->GetData(0) == nullptr) {
```

Comment:
See the comment removed in 6d16048da634eaf2254250262b9016ab69e1f57b, I don’t think `0` will be available unconditionally. You can probably just use `3` again, but in any case, this should be defined as a macro/constant somewhere.

You could probably use `node::Environment::GetCurrent(isolate)` to access the current `node::Environment`, which is a grab-bag of information for cases like these. The downside of that is that that would make the contents of this file quite a bit more Node-specific.